pkg/utils/utils.go

```diff
@@ -531,7 +530,7 @@ func getHealthServerResponse(conn *grpc.ClientConn) (*healthpb.HealthCheckRespon
 
 // CheckGrpcServerStatus will check gRPC server status using gRPC health check
 func CheckGrpcServerStatus(target string, log *log.Entry, grpcDial grpcDialType) (bool, error) {
-	conn, err := grpcDial(target, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	conn, err := grpcDial(target)
```

Comment:
I think you should call getCredentials() here too.
  
credentials, err := getCredentialFunc()
if err != nil {
        s.log.WithError(err).Error("error getting gRPC client credentials to connect to backend")
        return out, nil
    }
    conn, err := grpcDial(managerAddr, grpc.WithTransportCredentials(credentials))

Reply:
missed this one here. Added fix in latest commit